frontend/tests/controllers-test/challengeListCtrl.test.js

```diff
@@ -0,0 +1,282 @@
+                expect(utilities.storeData).toHaveBeenCalledWith("challengeCreator", vm.challengeCreator);
+            }
+        });
+
+        it('ongoing challenge backend error `challenges/challenge/present`', function () {
+            isPresentChallengeSuccess = false; 
+            isUpcomingChallengeSucess = null;
+            isPastChallengeSuccess = null;
+            errorResponse = {
+                error: 'error'
+            };
+            vm = createController();
+            expect(utilities.hideLoader).toHaveBeenCalled();
+        });
+
+        it('when no upcoming `challenges/challenge/present`challenge found `challenges/challenge/future`', function () {
+            isUpcomingChallengeSucess = true;
+            isPresentChallengeSuccess = true;
+            isPastChallengeSuccess = null;
+            successResponse = {
+                results: []
+            };
+            vm = createController();
+            expect(vm.upcomingList).toEqual(successResponse.results);
+            expect(vm.noneUpcomingChallenge).toBeTruthy();
+        });
+
+        it('check description length and calculate timezone of upcoming challenge `challenges/challenge/future`', function () {
+            isUpcomingChallengeSucess = true;
+            isPresentChallengeSuccess = true;
+            isPastChallengeSuccess = null;
+            successResponse = {
+                results: [
+                    {
+                        id: 1,
+                        description: "the length of the upcoming challenge description is greater than or equal to 50",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Fri June 12 2018 22:41:51 GMT+0530",
+                        end_date: "Fri June 12 2099 22:41:51 GMT+0530"
+                    },
+                    {
+                        id: 2,
+                        description: "random description",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Sat May 26 2015 22:41:51 GMT+0530",
+                        end_date: "Sat May 26 2099 22:41:51 GMT+0530"
+                    }
+                ]
+            };
+            vm = createController();
+            expect(vm.upcomingList).toEqual(successResponse.results);
+            expect(vm.noneUpcomingChallenge).toBeFalsy();
+
+            var timezone = moment.tz.guess();
+            for (var i in vm.upcomingList) {
+                if (vm.upcomingList[i].description.length >= 50) {
+                    expect(vm.upcomingList[i].isLarge).toEqual("...");
+                } else {
+                    expect(vm.upcomingList[i].isLarge).toEqual("");
+                }
+                var offset = new Date(vm.upcomingList[i].start_date).getTimezoneOffset();
+                expect(vm.upcomingList[i].start_zone).toEqual(moment.tz.zone(timezone).abbr(offset));

```

Comment:
Same thing here too, move `moment.tz.zone(timezone)` to before for since timezone isn't changing while you are trying to do your assertions.

Reply:
same here